lib/plugins/aws/deployFunction/tests/index.js

```diff
@@ -0,0 +1,245 @@
+    serverless.service.environment = {
+      vars: {},
+      stages: {
+        dev: {
+          vars: {},
+          regions: {
+            'us-east-1': {
+              vars: {},
+            },
+          },
+        },
+      },
+    };
+    serverless.service.functions = {
+      first: {
+        handler: true,
+      },
+    };
+    const options = {
+      stage: 'dev',
+      region: 'us-east-1',
+      function: 'first',
+    };
+    serverless.init();
+    awsDeployFunction = new AwsDeployFunction(serverless, options);
+  });
+
+  describe('#constructor()', () => {
+    it('should have hooks', () => expect(awsDeployFunction.hooks).to.be.not.empty);
+
+    it('should set the provider variable to "aws"', () => expect(awsDeployFunction.provider)
+      .to.equal('aws'));
+
+    it('should run promise chain in order', () => {
+      const checkIfFunctionExistsInServiceStub = sinon
+        .stub(awsDeployFunction, 'checkIfFunctionExistsInService').returns(BbPromise.resolve());
+      const checkIfFunctionIsDeployedStub = sinon
+        .stub(awsDeployFunction, 'checkIfFunctionIsDeployed').returns(BbPromise.resolve());
+      const zipFunctionStub = sinon
+        .stub(awsDeployFunction, 'zipFunction').returns(BbPromise.resolve());
+      const deployFunctionStub = sinon
+        .stub(awsDeployFunction, 'deployFunction').returns(BbPromise.resolve());
+
+      return awsDeployFunction.hooks['deploy:function:deploy']().then(() => {
+        expect(checkIfFunctionExistsInServiceStub.calledOnce).to.be.equal(true);
+        expect(checkIfFunctionIsDeployedStub.calledAfter(checkIfFunctionExistsInServiceStub))
+          .to.be.equal(true);
+        expect(zipFunctionStub.calledAfter(checkIfFunctionIsDeployedStub))
+          .to.be.equal(true);
+        expect(deployFunctionStub.calledAfter(zipFunctionStub))
+          .to.be.equal(true);
+
+        awsDeployFunction.checkIfFunctionExistsInService.restore();
+        awsDeployFunction.checkIfFunctionIsDeployed.restore();
+        awsDeployFunction.zipFunction.restore();
+        awsDeployFunction.deployFunction.restore();
+      });
+    });
+  });
+
+  describe('#checkIfFunctionExistsInService()', () => {
+    it('it should throw error if function is not provided', () => {
+      serverless.service.functions = null;
+      expect(() => awsDeployFunction.checkIfFunctionExistsInService()).to.throw(Error);
+    });
+  });
+
+  describe('#checkIfFunctionIsDeployed()', () => {
+    it('should check if the function is deployed', () => {
+      const getFunctionStub = sinon
+        .stub(awsDeployFunction.sdk, 'request').returns(BbPromise.resolve());
+
+      awsDeployFunction.functionName = 'first';
+
+      return awsDeployFunction.checkIfFunctionIsDeployed().then(() => {
+        expect(getFunctionStub.calledOnce).to.be.equal(true);
+        expect(getFunctionStub.calledWith(
+          awsDeployFunction.options.stage, awsDeployFunction.options.region)
+        );
```

Comment:
could you also make sure that the correct AWS SDK method and service are being passed? I see here that you're just checking we're passing stage and region args, but we actually also pass the AWS Service Name and Service Method